### PR TITLE
rdhs blog piece preview checks

### PR DESCRIPTION
**Pull request by OJWatson**

### CODE REVIEW

**Comment by OJWatson** [comment]
Hi there, 

Glad you like the overview. It's possibly a bit long and i'm still tweaking it a little, but just wanted to see what it looked like in netlify. I'll make sure to let you know here when i've finished editing it (hopefully by the end of tonight). 

Thanks again and sorry for how long it's taken me to write this.

**Comment by OJWatson** [comment]
Cool, i've finished tweaking, and looks good to me. Thanks again for the opportunity to write something.

File: content/blog/2019-01-31-rdhs.md [hunk 1]
```diff
@@ -0,0 +1,787 @@
...
+to do something properly. I had also at this point been keen to start using `testhat`
+within my work as I had been told it would save me time in the future, and up till
+that point I hadn't found a good case to get to grips with it (mainly writing code
+on my own, that was never very big and was only used by myself). And so we started
+writing `rdhs`, which was accepted by `ropensci` and `CRAN` in December 2018.
+
+### Package overview
+
+>
+Disclaimer: A lot of the following section (the **API** and **Dataset Downloads**
```
Comment (stefaniebutland) [inline content/blog/2019-01-31-rdhs.md]: Rather than replicating the vignette contents here, perhaps you could list what is covered in the vignette and link to it. In the post you can narrate your way briefly through the intriguing example without giving all the details so that people can see the potential of the package. 

this is good: "For example, to find out the trends in antimalarial use in Africa, and see if perhaps antimalarial prescription has decreased after RDTs were introduced (assumed 2010)." with the plot.

and this is a good end of that section:
"We now have managed to go from our opening question [make sure this is crystal clear at beginning of example] to a finalised data set that we can use going forwards for any downstream analysis (and hopefully it didn’t take that long to do it!)"

File: content/blog/2019-01-31-rdhs.md [hunk 2]
```diff
@@ -0,0 +1,787 @@
...
+---
+
+We now have managed to go from our opening question to a finalised data set that
+we can use going forwards for any downstream analysis (and hopefully it didn't
+take that long to do it!)
+
+### Ramblings after my first completed package
+
+Clichéd but the process of actually writing a package, and all that entailed, 
+was a real highlight. I had made R packages befor, but I had never done everything that a
```
Comment (stefaniebutland) [inline content/blog/2019-01-31-rdhs.md]:
```suggestion
was a real highlight. I had made R packages before, but I had never done everything that a 
```
fixed typo

File: content/blog/2019-01-31-rdhs.md [hunk 3]
```diff
@@ -0,0 +1,787 @@
...
+We now have managed to go from our opening question to a finalised data set that
+we can use going forwards for any downstream analysis (and hopefully it didn't
+take that long to do it!)
+
+### Ramblings after my first completed package
+
+Clichéd but the process of actually writing a package, and all that entailed, 
+was a real highlight. I had made R packages befor, but I had never done everything that a 
+good R package should have (tests, effective continuous integration, full documentation, 
+a `pkgdown` website, cntribution and code of conduct guides. and so on). One particular
```
Comment (stefaniebutland) [inline content/blog/2019-01-31-rdhs.md]:
```suggestion
a `pkgdown` website, contribution and code of conduct guides, and so on). One particular 
```

fix typos

File: content/blog/2019-01-31-rdhs.md [hunk 4]
```diff
@@ -0,0 +1,787 @@
...
+again for taking on this project. He definitely helped drive it over the finish line, 
+and it was nice to have a glimpse at what working as a developer would look like if 
+I decide to leave pure academia. 
+
+There were also a few things that before I started writing `rdhs`
+I knew I would have to figure out but I didn't have a clue where to start, and for
+which repeated googling didn't eventually help with. Fortunately, I work in the 
+same department as [Rich FitzJohn](https://twitter.com/rgfitzjohn), 
+so it was great having someone to point me in the right direction. The following, 
+are 3 of the things that I genuinely had no idea how to do before, so I thought I'd
```
Comment (stefaniebutland) [inline content/blog/2019-01-31-rdhs.md]:
```suggestion
are three of the things that I genuinely had no idea how to do before, so I thought I'd
```

3 -> three

File: content/blog/2019-01-31-rdhs.md [hunk 5]
```diff
@@ -0,0 +1,787 @@
...
+#### 1. Logging into a website from R
+
+The DHS website has a download manager that you can use to select surveys you want to 
+download, and it will auto generate a list of URLs in a text file. When I saw this, I thought
+this would be great for creating a database of what data sets and the URLs a user's login details
+can give them, which can then be cached so that `rdhs` knows whether you can download a data set
+or not. The only problem is, that to download those data sets you need to be logged in, and you 
+also need to be logged in to get to the download manager. For me, I didn't know how to translate
+being "logged in" into R code, or even what that looked like. But turns out it wasn't too bad
+after being shown where to start looking by Rich.
```
Comment (stefaniebutland) [inline content/blog/2019-01-31-rdhs.md]:
```suggestion
after being shown by Rich where to start looking. 
```

File: content/blog/2019-01-31-rdhs.md [hunk 6]
```diff
@@ -0,0 +1,787 @@
...
+in the page to generate the URL with all the download links in. 
+
+#### 2. Caching API results from a changing API
+
+We wanted to be able to cache a user's API request for them locally when designing 
+`rdhs`. We felt this was important as it would reduce the burden on the API itself, 
+as well as enable researchers who were without internet (e.g. currently working in 
+the field), the ability to still access previous API requests. However, designing
+something neat that would be easy to respond to changes in the API version would
+I thought would be outside my skill set.
```
Comment (stefaniebutland) [inline content/blog/2019-01-31-rdhs.md]:
```suggestion
I thought be outside my skill set. 
```

File: content/blog/2019-01-31-rdhs.md [hunk 7]
```diff
@@ -0,0 +1,787 @@
...
+#### 2. Caching API results from a changing API
+
+We wanted to be able to cache a user's API request for them locally when designing 
+`rdhs`. We felt this was important as it would reduce the burden on the API itself, 
+as well as enable researchers who were without internet (e.g. currently working in 
+the field), the ability to still access previous API requests. However, designing
+something neat that would be easy to respond to changes in the API version would
+I thought would be outside my skill set. 
+
+Again, enter Rich and this time with his package [storr](https://cran.r-project.org/web/packages/storr/index.html).
```
Comment (stefaniebutland) [inline content/blog/2019-01-31-rdhs.md]:
```suggestion
Again, enter Rich and this time with his package [`storr`](https://cran.r-project.org/web/packages/storr/index.html).
```

please put package names in backticks, even when they are linked to something

File: content/blog/2019-01-31-rdhs.md [hunk 8]
```diff
@@ -0,0 +1,787 @@
...
+me to test the weird edge cases that started popping up related to certain data 
+sets. The first solution that I used for a few months was to set up environment 
+variables within `travis` itself, which could then be used to create a valid
+set of credentials. 
+
+![](/img/blog-images/2019-01-31-rdhs/travis_env_vars.png)
+
+This worked, however, it meant that I would have to write a lot of the `rdhs`
+functionality to use environment variables that were the user's email and password, 
+which felt wrong and quite clunky. All I wanted was to pass to Travis was a valid
```
Comment (stefaniebutland) [inline content/blog/2019-01-31-rdhs.md]:
```suggestion
which felt wrong and quite clunky. All I wanted was to pass to Travis a valid
```

File: content/blog/2019-01-31-rdhs.md [hunk 9]
```diff
@@ -0,0 +1,787 @@
...
+
+# read this tar in as binary data 
+dat <- readBin("rdhs.json.tar",raw(),file.size("rdhs.json.tar"))
+
+# encrypt the data using sodium and our key before saving it
+enc <- sodium::data_encrypt(msg = dat,key = key)
+saveRDS(enc,"rdhs.json.tar.enc")
+```
+
+This could then be included in the GitHub repository, and I could then
```
Comment (stefaniebutland) [inline content/blog/2019-01-31-rdhs.md]: 3 "then's" in 2 sentences :-)

File: content/blog/2019-01-31-rdhs.md [hunk 10]
```diff
@@ -0,0 +1,787 @@
...
+set up the key as a Travis environment variable to decrypt it. This decryption 
+step could then be written within my `.travis.yml` file, and would mean that all
+my tests had access to my login credentials in a secure way. 
+
+### Acknowledgements and Final Thoughts
```
Comment (stefaniebutland) [inline content/blog/2019-01-31-rdhs.md]: This section has a really nice genuine tone

File: content/blog/2019-01-31-rdhs.md [hunk 11]
```diff
@@ -0,0 +1,787 @@
...
+
+Firstly, I want to thank [Anna Krystalli](https://github.com/annakrystalli) for handling
+the review, and for being incredibly patient throughout, especially at the end as we
+were fixing the last authentication bug. Also many thanks to [Lucy McGowan](https://github.com/LucyMcGowan)
+and [Duncan Gillespie](https://github.com/dosgillespie) for taking the time to 
+review the package and for their input, which led to lots of improvements (and also
+linking the `add_line` function from `httr` was seriously helpful, and I've used 
+that function in lots of other my other work now). I also wanted to more broadly thank 
+the review process as a whole. Having the option to discuss the package and needed 
+solutions with the reviewers within a GitHub issues system is fantastic. It made the process
```
Comment (stefaniebutland) [inline content/blog/2019-01-31-rdhs.md]:
```suggestion
solutions with the reviewers within a [GitHub issues system](https://github.com/ropensci/software-review/issues/238) is fantastic. It made the process 
```

maybe even link to the review?

File: content/blog/2019-01-31-rdhs.md [hunk 12]
```diff
@@ -0,0 +1,787 @@
...
+Lastly, another big thank you [Jeff Eaton](https://github.com/jeffeaton) and 
+[Rich FitzJohn](https://github.com/richfitz), and also to the infectious
+disease epidemiology department at Imperial for providing a lot of really helpful 
+guinea pig testing of the numerous iterations of `rdhs`.
+
+---
+
+### Options to Contribute
+
+There are a few things that would be great to add in the future.
```
Comment (stefaniebutland) [inline content/blog/2019-01-31-rdhs.md]: For visual variety, you might put these in a numbered list rather than saying "first, second...", but no obligation to do this

File: content/blog/2019-01-31-rdhs.md [hunk 13]
```diff
@@ -0,0 +1,787 @@
...
+
+The first would be to add a suite of tools for doing spatial mapping. A lot of the
+time, people want to know what the prevalence of x is either at a fine spatial scale, 
+or grouped at administrative/county/state levels. `rdhs` helps provide the tools to
+get geolocated measures of x, and I think it would be a great next step to add
+a suite of mapping tools. It would be great if they could be used to either create a mesh 
+through these points (probably using `INLA`), or calculate survey weighted means at requested
+spatial scales or match them to a provided `SpatialPolygons` object. Related to this is
+it would be good to also link in the [Spatial Data Repository](https://spatialdata.dhsprogram.com/home/)
+from the DHS, so that users can easily download shape files for their analyses ([issue #71](https://github.com/ropensci/rdhs/issues/71).
```
Comment (stefaniebutland) [inline content/blog/2019-01-31-rdhs.md]:
```suggestion
from the DHS, so that users can easily download shape files for their analyses ([issue #71](https://github.com/ropensci/rdhs/issues/71)).
```

added missing parenthesis

File: content/blog/2019-01-31-rdhs.md (hunk 10 above)
Comment (stefaniebutland) [reply content/blog/2019-01-31-rdhs.md]: I think it would be better to have this section last. The sentiment is great, and "Options to Contribute" leaves the post with an abrupt end

**Comment by OJWatson** [comment]
@stefaniebutland  thanks so much for the kind words and the suggestions (have never used the github suggests feature before so double thanks for showing me that as it's really cool). 

I should have the revisions done by then so i'll go ahead and start making those changes

**Comment by stefaniebutland** [comment]
I'll have another look when you're finished making changes so please tag me then @OJWatson

File: content/blog/2019-01-29-rdhs.md [hunk 14]
```diff
@@ -0,0 +1,580 @@
...
+## NegativeTest PositiveTest 
+##            0            1
+```
+
+For more information about controlling how to extract data from your downloaded
+sections, see the [last section in the introduction vignette](https://ropensci.github.io/rdhs/articles/introduction.html#extract-variables-and-combining-datasets-for-pooled-multi-survey-analyses-). 
+
+---
+
+We now have managed to go from our initial request for data about the use of
```
Comment (stefaniebutland) [inline content/blog/2019-01-29-rdhs.md]: Can you add 1-2 sentences saying more specifically what you now have in the "a finalised data set" after these 5 steps? The plot in step 1 lures the reader with a really interesting research question, but after focusing on the technical steps, it's hard to know what we end up with. 

For the record, I'm realizing this is one of the most common things I see in draft blog posts about packages.

File: content/blog/2019-01-29-rdhs.md [hunk 15]
```diff
@@ -0,0 +1,580 @@
...
+helpful (eventually) was really great. With this in mind I wanted to thank [Jeff Eaton](https://twitter.com/eatonjw)
+again for taking on this project. He definitely helped drive it over the finish line, 
+and it was nice to have a glimpse at what working as a developer would look like if 
+I decide to leave pure academia. 
+
+There were also a few things that before I started writing `rdhs`
+I knew I would have to figure out but I didn't have a clue where to start, and for
+which repeated googling didn't eventually help with. Fortunately, I work in the 
+same department as [Rich FitzJohn](https://twitter.com/rgfitzjohn), 
+so it was great having someone to point me in the right direction. The following,
```
Comment (stefaniebutland) [inline content/blog/2019-01-29-rdhs.md]:
```suggestion
so it was great having someone to point me in the right direction. The following 
```
omit comma after "following"

File: content/blog/2019-01-29-rdhs.md [hunk 16]
```diff
@@ -0,0 +1,580 @@
...
+
+#### 2. Identify surveys relevant for further analysis
+
+You may, however, wish to do more nuanced analysis than the API allows. 
+The following 4 sections detail a very basic example of how to quickly 
+identify, download and extract datasets you are interested in.  
+
+Let's say we want to get all DHS survey data from the Democratic Republic of 
+Congo and Tanzania in the last 5 years (since 2013), which covers the use of 
+rapid diagnostic tests (RDTs) for malaria. To begin we'll interact with the
```
Comment (stefaniebutland) [inline content/blog/2019-01-29-rdhs.md]:
```suggestion
rapid diagnostic tests for malaria ("RDT" below). To begin we'll interact with the 
```
modify abbreviation text

File: content/blog/2019-01-29-rdhs.md [hunk 17]
```diff
@@ -0,0 +1,580 @@
...
+##    SurveyCharacteristicID SurveyCharacteristicName
+## 57                     96            Malaria - DBS
+## 58                     90     Malaria - Microscopy
+## 59                     89            Malaria - RDT
+## 60                     57          Malaria module 
+## 61                      8 Malaria/bednet questions
+```
+
+There are 87 different survey characteristics, with one specific survey 
+characteristic for Malaria RDTs. In this example we will use this to find the surveys
```
Comment (stefaniebutland) [inline content/blog/2019-01-29-rdhs.md]:
```suggestion
characteristic for Malaria rapid diagnostic tests (RDT). In this example we will use this to find the surveys 
```
add full text for abbrev

File: content/blog/2019-01-29-rdhs.md [hunk 18]
```diff
@@ -0,0 +1,580 @@
...
+*hv024* stores the regions for these 2 countries, and we probably want to keep all 
+the labels, which we can do by using the `labels` argument:
+
+
+```r
+# lets try concatenating the hv024
+better_bound <- rbind_labelled(extract, labels = list("hv024"="concatenate"))
+```
+We could also specify new labels for a variable. For example, imagine the two 
+datasets encoded their RDT responses differently, with the first one as
```
Comment (stefaniebutland) [inline content/blog/2019-01-29-rdhs.md]:
```suggestion
datasets encoded their rapid diagnostic test responses differently, with the first one as 
```
spell out RDT

File: content/blog/2019-01-29-rdhs.md [hunk 19]
```diff
@@ -0,0 +1,580 @@
...
+##            0            1
+```
+
+For more information about controlling how to extract data from your downloaded
+sections, see the [last section in the introduction vignette](https://ropensci.github.io/rdhs/articles/introduction.html#extract-variables-and-combining-datasets-for-pooled-multi-survey-analyses-). 
+
+---
+
+We now have managed to go from our initial request for data about the use of 
+rapid diagnostic tests (RDTs) for malaria to a finalised data set that
```
Comment (stefaniebutland) [inline content/blog/2019-01-29-rdhs.md]:
```suggestion
rapid diagnostic tests for malaria to a finalised data set that
```
remove abbrev

File: content/blog/2019-01-29-rdhs.md [hunk 20]
```diff
@@ -0,0 +1,580 @@
...
+#### What is the DHS Program
+
+The [Demographic and Health Survey (DHS) Program](https://www.dhsprogram.com) 
+has collected and disseminated population survey data from 
+over 90 countries for over 30 years. This amounts to over 400 
+surveys that give representative data on health indicators, which in
+many countries provides the key data that mark progress towards targets such as 
+the [Sustainable Development Goals (SDGs)](https://sustainabledevelopment.un.org/sdgs). In addition,
+DHS survey data has been used to inform health policy such as detailing trends in child mortality[^1] 
+and characterising the distribution of malaria control interventions in Africa in order to map the
```
Comment (stefaniebutland) [inline content/blog/2019-01-29-rdhs.md]: You have used "Malaria" and "malaria". Check whole doc for consistency (maybe capitalization is warranted sometimes but if so, that's not clear)

**Comment by stefaniebutland** [comment]
@OJWatson I see you're making edits so please tag me here when you're finished

File: content/blog/2019-01-29-rdhs.md (hunk 14 above)
Comment (OJWatson) [reply content/blog/2019-01-29-rdhs.md]: I've added a bit more here and have tried to reframe the end of the package overview, to try and remind the reader why we are interested in building these data sets in the first place. Thanks for picking this up.

File: content/blog/2019-01-29-rdhs.md (hunk 20 above)
Comment (OJWatson) [reply content/blog/2019-01-29-rdhs.md]: changed to lower case throughout

File: content/blog/2019-01-29-rdhs.md [hunk 21]
```diff
@@ -0,0 +1,580 @@
...
+```
+
+### API
+
+#### 1. Access standard indicator data via the API
+
+The DHS program has published an API that gives access to 12
+different data sets. Each API endpoint represents one of the 12 data sets 
+(e.g. https://api.dhsprogram.com/rest/dhs/tags), and can be accessed using the `dhs_<>()` functions. For 
+more information about this
```
[the remark this answers is not in the document]
Comment (OJWatson) [reply content/blog/2019-01-29-rdhs.md]: fixed

**Comment by OJWatson** [comment]
@stefaniebutland I've just finished adjusting the text to address your last comments. let me know if there is anything else and sorry for the late reply in the day.

**Comment by stefaniebutland** [comment]
checking now @OJWatson 
no apology necessary - your responses have been quite timely!

File: content/blog/2019-01-29-rdhs.md (hunk 14 above)
Comment (stefaniebutland) [reply content/blog/2019-01-29-rdhs.md]: this is great thanks

**Comment by OJWatson** [comment]
Have changed. Thanks again
